docs/proposals/improve-addons.md

```diff
@@ -0,0 +1,350 @@
+
+In order to track available addons, the AddonCatalogueDistributor makes use of AddonCatalogue CRs
+
+### AddonCatalogue CR
+
+- contains a list of all addons
+- each addon comes in multiple versions
+- each version has
+  - template
+    - a rendering method. Possible methods are (additional methods can easily be added in the future):
+      - go-template → same as what we currently offer in addons
+      - helm → rendering of helm charts
+    - formSpec → Similar to the current formSpec. It is important to know that this value overwrites any other value store (e.g. helm's values.yaml):
+      - displayName →Name in the UI
+      - internalName →Templating reference. Allows for nested values using the dot-syntax. For example, if you want to overwrite a values.yaml file, you could do `internalName: spec.replicas`
+      - type →Type of the field. We propose that references for now can only be primary types (number, string, ...), but do not allow for nested types (e.g. objects). This makes rendering them a lot more convenient
+      - (optional) helpText →helpText to be displayed in the UI
+  - a source from where to pull the data from. Possible sources are:
+    - docker-image →same as we currently have. A Docker image contains manifests for one or multiple addons and is being pulled by the controller
+    - git → a git repository from which to pull the manifests. This was considered to be especially useful by Kubermatic PS who work directly with customers
+    - helm → a helm repository to pull charts from. Only compatible with rendering method helm
+  - constraints. These describe conditions that must apply for an add-on to be compatible with a user cluster. Possible constraints are:
+    - kkp-version →semVer range that describes compatible KKP versions
+    - k8s-version →semVer range that describes compatible k8s versions
+    - *tbd: Are there any other relevant constraints?*
+  - values → These are values that can be used for overwriting defaults. We think this field will be required for CoreAddons if a customer is using a private registry. So addons can be configured
+
+Note: The current implementation of go-templating allows to declare a `RequiredResourceTypes` field, to handle dependencies. We propose to keep this as is for go-templating, but not have it for helm. The reason is that helm offers its own dependency management and we think that building an additional dependency mechanism on top will drastically increase complexity. As a result, this field will be moved into under go-template method and not on the general addon level.
+
+An example CR could like this:
+
+```yaml
+apiVersion: kubermatic.k8c.io/v1
+kind: AddonCatalogue
+metadata:
+ name: custom-catalogue
+spec:
+ addons:
+     - prometheus:
+            description: "The Prometheus Node Exporter exposes a wide variety of hardware- and kernel-related metrics."
+            logoFormat: "svg+xml"
+            logo: # logo must be base64 encoded.
```

Comment:
Size of it may be problematic if we will store the logos together. In the @kubermatic/sig-api @kubermatic/sig-ui we were considering a change here, perhaps we should aim to host these images somewhere.

Reply:
good point. I have added logos as a non-goal for now, so we can move forward with the topic. Good news is: With the current architecture, we can add logos later on without any problems; be it base64 encoded or hosted somewhere else